2024/planning-process.md

```diff
@@ -0,0 +1,134 @@
+
+### Elimination Round 3: Prioritization
+
+The official list of Focus Area proposals is updated to reflect the elimination of proposals not chosen by any organization. Each organization now has 20 days to review the list of remaining proposals and make their Round 3 decisions.
+
+In this round, each organization ranks the proposals in order of importance from most to least. And submits any vetos they may wish to exercise.
+
+Any proposal that gets a veto is immediately eliminated.
+
+Each proposal gets assigned points from each organization, according to their rank, and an ordered list of all remaining proposals is created. Where `X` is number of proposals that are ranked, and `Y` is the ranking a particular proposal gets from a certain organization, then that proposal gets `X-Y`  points from that organization. Once all 6 rankings are in for that proposal, the total points from all six organizations is known. If that proposal is not eliminated by a veto, then it has a certain number of points, and can be listed in order among the other proposals.
+
+For example, if there are 20 remaining proposals, each organization will choose a number 1, number 2, down to number 20. Let’s say a proposal for flying cars gets the following ranks: 1, 5, 7, 10, 14, 20. That will give it the following points: 20, 15, 13, 10, 6, 1 = 65 points total.
+
+(If there appear to be quite a few proposals remaining in this round, the group can decide to only ask for each organization to submit their Top `X` number, while rating the rest zero. For example, if 50 proposals remain at this point, it likely makes sense to only submit a Top 25, and to not worry about ranking 26-50 in a specific order.)
+
+After Round 3, every remaining proposal can be listed in order, with the group’s highest priority at the top, and the lowest at the bottom.
+
+NOTE: in the past, “Exclude” was a way to express many different kinds of sentiments — including: does not meet the minimum qualifications, or this is low priority to us. We expect “veto” to be used much less than “exclude”, because there are other ways to express those sentiments. Vetoing is specifically for objecting to a particular proposal being included *at all,* no matter how highly ranked other organizations might find it. It’s for expressing objection to a particular web technology, perhaps to something an organization has already expressed an objection to in the web standards process or otherwise has an objection to implementing.
+
+## Interop 2023 Evaluation
+
+During the same period organizations are making Round 3 decisions, the organizing committee should evaluate the status of Interop 2023, and decide which Focus Areas will be retired, and which will be carried over.
+
+\[XXX]
+
+* * *
+
+# Decision point — what to do about Interop 2023?
+
+## There seem to be 3 options:
```

Comment:
OK, so based on the discussion, here's an alternative option:

1. Do an (early) post-mortem for all Interop 2023 proposals. File an issue on GitHub for each one, detailing which tests are not yet interoperable (if any).
1. Encourage participants to comment on the focus area. For example they might want to include:
    * Information about whether significant changes are planned/expected in the near future in their implementations.
    * Additional context they can share around remaining failures (e.g. reasons that they turned out to be lower priority or harder to fix).
    * Any additional coverage that's missing from the focus area and should be included if the focus area is carried forward.
1. Given this information, we compile a list of candidate focus areas to carry forward. This would include any where we believe there is still meaningful Interop work to be completed, either because we can see the remaining failures are meaningful (i.e. it's not a collection of unlikely edge cases), or because we believe that the tests/scope of the focus area should be increased.
1. At proposal selection time, we work from both the list of potential carry-over and the list of new proposals in order to fill the project out to the desired size. There is no commitment to take everything on the potential carry-over list, but there is an assumption that items will be carried over if they are legitimately seen as ongoing priority areas for improving Interop.

This has the following benefits over any of the existing proposals:
* It encourages us to make an evidence-based decision on the impact of remaining failures in existing focus areas, rather than assuming some fixed numerical cut off, or making an ad-hoc decision based on gut feelings.
* It explicitly allows for the possibility of these focus areas being expanded with new tests that may cover additional scenarios which weren't included in 2023.
* It allows us to provide explicit priority to carryover proposals, without requiring that any carryover always outranks new proposals, as in the current option 1.

Of course this would still be subject to the usual consensus-based process for final decisions.

Reply:
I think we need to get agreement on the more fundamental disagreement first: are we requiring consensus to carryover *or* are we require consensus to retire? Because what we should do process-wise likely depends on that decision.

Reply:
I don't think it makes sense to talk about consensus except in the positive sense i.e. complete agreement rather than complete disagreement.

If we don't have consensus that something actively should be in Interop 202x then we don't have consensus to publish that iteration of the metric.

Reply:
I can see the argument for both options:
- Requiring consensus to carryover makes sense since any objection means someone organization isn't happy with the metric, and over time this can threaten the viability of the project.
- Requiring consensus to retire makes sense since we should finish what we already agreed to, and not relitigate.

But I'm most convinced by the first argument. I don't think there's a significant risk of us dropping things that are still very important with this setup.